static/src/javascripts/projects/common/modules/commercial/ads/user-ad-feedback.js

```diff
@@ -10,39 +10,27 @@ define([
     config
 ) {
 
-    function userAdFeedback(pagePath, adSlotId, creativeid, reason) {
-        // TODO: This. Better.
-        if (reason !== 'ad-feedback-menu-opened') {
-            return recordUserAdFeedback(pagePath, adSlotId, creativeid, reason);
-        } else {
-            return recordUserAdFeedback(pagePath, adSlotId, creativeid, 'ad-feedback-menu-opened');
-        }
-    }
-
-    function recordUserAdFeedback(pagePath, adSlotId, creativeId, feedbackType) {
+    return function recordUserAdFeedback(pagePath, adSlotId, creativeId, feedbackType) {
         var feedbackUrl = 'https://j2cy9stt59.execute-api.eu-west-1.amazonaws.com/prod/adFeedback';
         var stage = config.page.isProd ? 'PROD' : 'CODE';
-        var data = '{"stage":"'+stage+'", "adPage":"'+pagePath+'", "adSlotId":"'+adSlotId+'", "adCreativeId":"'+creativeId+'", "adFeedback":"'+feedbackType+'"}';
+        var data = JSON.stringify({
+            stage: stage,
+            adPage: pagePath,
+            adSlotId: adSlotId,
+            adCreativeId: creativeId.toString(),
+            adFeedback: feedbackType
+        });
         return fetch(feedbackUrl, {
             method: 'post',
             body: data,
             mode: 'cors'
-        }).then(function() {    // ignore the response
-        }).catch(function() {    // ignore any errors
-        }).then(function() {    // we're complete - update the UI
-            if (feedbackType !== 'ad-feedback-menu-opened') {
-                var adSlot = bonzo(qwery('#' + adSlotId));
-                adSlot.each(function (ad) {
-                    for (var i = 0; i < ad.children.length; i++) {
-                        if (ad.children[i].classList.contains('ad-slot__label')) {
-                            ad.children[i].innerHTML = 'Advertisement (Thanks for your feedback)';
-                        }
-                    }
-                });
-            }
-        });
-    }
+        }).then(onComplete(adSlotId, feedbackType), function(){});
```

Comment:
Nearly there! You need to pass the same callback twice, so that it is called no matter what happens, a rejection or a resolve. And you also need to wrap it in a lambda:
```
.then(
  function() { return onComplete(adSlotId, feedbackType); },
  function() { return onComplete(adSlotId, feedbackType); }
)
```